components/board/pinwrappers/analogs.go

```diff
@@ -29,7 +30,7 @@ type AnalogSmoother struct {
 }
 
 // SmoothAnalogReader wraps the given reader in a smoother.
-func SmoothAnalogReader(r AnalogReader, c AnalogReaderConfig, logger logging.Logger) *AnalogSmoother {
+func SmoothAnalogReader(r board.AnalogReader, c board.AnalogReaderConfig, logger logging.Logger) *AnalogSmoother {
```

Comment:
Discussion topic: the `board.AnalogReaderConfig` describes the samples per second and number of seconds to smooth over. Perhaps it's really a `SmoothAnalogReaderConfig` that belongs in this package instead of `board`. In the pi4, genericlinux, and numato boards, the config is predominantly used by passing it to the SmoothAnalogReader constructor, rather than something with a "raw" analog reader.

...then again, perhaps this is part of the further cleanup in a future PR.

Reply:
My thinking: yes, and we we eventually use the AnalogSmoother only in the numato driver and yeet the MCP3008 out of existence in genericlinux and the pi4 drivers.

Reply:
@sguequierre is in the middle of documenting the MCP stuff for all the different genericlinux boards. If we want to yeet it, we should move quickly on that, before users find out about it. :smiling_imp: